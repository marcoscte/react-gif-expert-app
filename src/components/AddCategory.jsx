import React, { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChange = (event) => {
        // console.log(event.target.value);
        setInputValue(event.target.value);
    }
    const onSubmit = (event) => {

        event.preventDefault();
        const valueClean = inputValue.trim();
        if(valueClean.length<=1) return;
        //setCategories(categories =>[inputValue, ...categories]);
        onNewCategory(valueClean);
        setInputValue('');

    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                name=""
                id="addCategoryText"
                placeholder='Search GIF'
                value={inputValue}
                onChange={onInputChange}
            />


        </form>
    )
}
