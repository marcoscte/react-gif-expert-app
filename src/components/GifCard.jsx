import React from 'react';
import PropTypes from 'prop-types'; 

export const GifCard = ({title, url, image, id}) => {

  return (
    <div className='card' key={id}>{title}
        <img src={url} alt={title} />
    </div>
  )
}

GifCard.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired
};
