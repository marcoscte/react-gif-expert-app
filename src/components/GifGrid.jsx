import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifCard } from './GifCard';
import PropTypes from 'prop-types'; 


export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category);

  return (
    <>
        <h3>{category}</h3>
        {
            isLoading && (<h2>Cargando...</h2>)
        }
        <div className='card-grid'>
            {images.map(image =>{
                return(                    
                    <GifCard title = {image.title} url = {image.url} key={image.id} image={image}/>
                );
            })}

        </div>
        
    </>
  )


}
GifGrid.propTypes = {
    category:PropTypes.string.isRequired,
}
