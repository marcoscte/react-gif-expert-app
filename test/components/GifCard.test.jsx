import { GifCard } from "../../src/components/GifCard";
import { render, screen} from '@testing-library/react';

describe('Pruebas en componente GifCard ', () => { 
    const title = 'Naruto';
    const url = 'https://media3.giphy.com/media/mlCb3AjEE6N4Q/giphy.gif?cid=2405e2ba4gt6rxocffmawjl07dh3z4kzzie6dhkacz98wxv1&rid=giphy.gif&ct=g';

    test('testear el snapshot', () => { 
        const {container} = render (<GifCard key={'123'} title={title} url={url}/>);
        expect(container).toMatchSnapshot();
     });


     test('debe mostrar la imagen con el url y el ALT indicado', () => { 
        render(<GifCard key={'123t'} title={title} url={url}/>);

        // expect(screen.getByRole('img').src).toEqual(url);
        // expect(screen.getByRole('img').alt).toEqual(title);

        const{src,alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);



      });

      test('Debe de mostrar en titulo en el componente', () => { 

        render(<GifCard key={'123t'} title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
       })
 })