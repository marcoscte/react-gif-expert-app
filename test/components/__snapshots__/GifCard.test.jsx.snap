// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifCard  testear el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    Naruto
    <img
      alt="Naruto"
      src="https://media3.giphy.com/media/mlCb3AjEE6N4Q/giphy.gif?cid=2405e2ba4gt6rxocffmawjl07dh3z4kzzie6dhkacz98wxv1&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
